jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/HttpClientStreamTest.java

```diff
@@ -1149,9 +1150,17 @@ public void close() throws IOException
 
         assertTrue(completeLatch.await(5, TimeUnit.SECONDS));
         assertTrue(closeLatch.await(5, TimeUnit.SECONDS));
-        assertTrue(serverDemandLatch.await(5, TimeUnit.SECONDS));
-        assertTrue(Content.Chunk.isFailure(lastChunk.get(), true));
-        assertInstanceOf(IOException.class, lastChunk.get().getFailure());
+        // Since the connector was stopped, there is no guarantee that
+        // the pending demand got serviced with an error chunk.
```

Comment:
The problem lies in `HTTP2ServerConnection.onStreamFailure()` which does the following:

```java
Runnable task = channel.onFailure(failure, callback);
if (task != null)
{
    // We must dispatch to another thread because the task
    // may call application code that performs blocking I/O.
    offerTask(task, true); // true means call strategy.dispatch()
}
```

Since it is the `task` returned by `channel.onFailure()` that eventually services the demand, there is no guarantee that it's going to be executed as there is a race condition between the strategy's thread executing the queued task and the strategy being stopped because it is part of the connector being stopped.